hbase-server/src/main/java/org/apache/hadoop/hbase/replication/master/ReplicationLogCleaner.java

```diff
@@ -92,12 +96,20 @@ public boolean apply(FileStatus file) {
   }
 
   @Override
-  public void setConf(Configuration config) {
-    // Make my own Configuration.  Then I'll have my own connection to zk that
-    // I can close myself when comes time.
-    Configuration conf = new Configuration(config);
+  public void init(Map<String, Object> params) {
+    super.init(params);
     try {
-      setConf(conf, new ZKWatcher(conf, "replicationLogCleaner", null));
+      if (MapUtils.isNotEmpty(params)) {
+        Object master = params.get(HMaster.MASTER);
+        if (master != null && master instanceof HMaster) {
+          zkw = ((HMaster) master).getZooKeeper();
+          shareZK = true;
```

Comment:
Better set this flag to true by default, and in the below `if (zkw == null)`, set it to false. Without any other assumptions, it is possible that HMaster.getZooKeeper returns null and we will create a new ZKWatcher but shareZK is false then we will not close it...

Reply:
yes，it is bettetr. But it is impossible that HMaster.getZooKeeper returns null，and if zk of hmaster is null, master must be abort.